bitnami/thanos/templates/receive/statefulset.yaml

```diff
@@ -71,13 +71,17 @@ spec:
             - --grpc-address=0.0.0.0:10901
             - --http-address=0.0.0.0:10902
             - --remote-write.address=0.0.0.0:19291
-            - --receive.replication-factor=1
+            - --receive.replication-factor={{ .Values.receive.replicationFactor }}
             - --objstore.config=$(OBJSTORE_CONFIG)
             - --tsdb.path=/var/thanos/receive
             - --label={{ .Values.receive.replicaLabel }}="$(NAME)"
             - --label=receive="true"
             - --tsdb.retention=15d
-            - --receive.local-endpoint=127.0.0.1:10901
+            {{- if and (eq .Values.receive.replicaCount 1) .Values.receive.config }}
+            - --receive.local-endpoint=127.0.0.1:{{ .Values.receive.service.grpc.port }}
+            {{- else }}
+            - --receive.local-endpoint=$(NAME).thanos-receive{{ if .Values.receive.service.additionalHeadless }}-headless{{ end }}.$(NAMESPACE).svc.{{ .Values.clusterDomain }}:10901
```

Comment:
In this case, I do see the value of using the service port (instead of hardcoding it)

Reply:
So just hardcode I will hardcode it here.

Reply:
I think both should be hardcoded.
Because its the fqdn for the pod itself. It will allways listen on that port unless we make it configurable.

Reply:
as this port is not configurable we should not change it.